src/hues.rs

```diff
@@ -12,81 +16,102 @@ macro_rules! make_hues {
         ///also have some surprising effects if it's expected to act as a
         ///linear number.
         #[derive(Clone, Copy, Debug, Default)]
-        pub struct $name(f32);
+        pub struct $name<T:Float = f32>(T);
 
-        impl $name {
+        impl<T:Float> $name<T> {
             ///Create a new hue from radians, instead of degrees.
-            pub fn from_radians(radians: f32) -> $name {
-                $name(radians * 180.0 / PI)
+            pub fn from_radians(radians: T) -> $name<T> {
+                $name(radians * T::from(180.0).unwrap() / T::from(PI).unwrap())
             }
 
             ///Convert the hue to radians.
-            pub fn to_radians(self) -> f32 {
-                normalize_angle(self.0) * PI / 180.0
+            pub fn to_radians(self) -> T {
+                normalize_angle(self.0) * T::from(PI).unwrap() / T::from(180.0).unwrap()
+            }
+
+            ///Returns the hue value as a f32 or f64 as needed
+            pub fn to_float(self) -> T {
+                normalize_angle(self.0)
             }
         }
 
-        impl From<f32> for $name {
-            fn from(degrees: f32) -> $name {
+        impl<T:Float> From<T> for $name<T> {
+            fn from(degrees: T) -> $name<T> {
                 $name(degrees)
             }
         }
 
-        impl Into<f32> for $name {
+        // TO-DO:
+        // Error: with  conflicting implementations of trait `core::convert::Into<_>` for type  `hues::RgbHue<_>`
+        // Error: conflicting implementation in crate `core`
+        // impl<T:Float> Into<T> for $name<T> {
+        //     fn into(self) -> T {
+        //         normalize_angle(self.0)
+        //     }
+        // }
```

Comment:
I don't think that will be possible at all before specialization arrives. In any case, it would be great if we can keep the to-dos in issues instead of comments :smile: